client/state/partner-portal/licenses/hooks/use-billing-dashboard-query.ts

```diff
@@ -99,12 +99,30 @@ export default function useBillingDashboardQuery< TError = unknown >(
 	const dispatch = useDispatch();
 	const activeKeyId = useSelector( getActivePartnerKeyId );
 
-	return useQuery< APIBilling, TError, Billing >(
+	const response = useQuery< APIBilling, TError, Billing >(
 		[ 'partner-portal', 'billing-dashboard', activeKeyId ],
 		queryBillingDashboard,
 		{
 			select: selectBillingDashboard,
-			onError: () => {
+			onError: ( error ) => {
+				// We wish to handle the "no billing invoice available" response differently
+				// from hard errors. We want this because the response itself did not encounter
+				// any errors but the upcoming invoice has not been created yet.
+				if ( error.hasOwnProperty( 'code' ) && 'no_billing_invoice_available' === error.code ) {
+					dispatch(
+						plainNotice(
+							translate(
+								'We have not started calculating your upcoming invoice yet. The statistics might not be correct.'
```

Comment:
I think we can improve the wording. How about this as an example:
```suggestion
								'Your upcoming invoice is being prepared and will be available soon.'
```

Reply:
I like the updated wording as since we use `-` instead of `0` as the fallback/default value we no longer need the `The statistics might not be correct.` disclaimer anyway.